[LLM] Pixtral example (#3938)
skypilot-org/skypilot: llm/pixtral/README.md

> # Deploy Pixtral with One Click on Any Cloud or Kubernetes

Reviewer: Deploy Pixtral On Kubernetes or Your Own Cloud

  ███████████████████░░░░░█████████████████████░░░░███████████████████░░░░░
  ███████████████████░░░░░████████████████████░░░░░███████████████████░░░░░
  ███████████████████░░░░░░███████████████████░░░░░███████████████████░░░░░
  ███████████████████░░░░░░██████████████████░░░░░░███████████████████░░░░░
  ███████████████████░░░░░░░█████████████████░░░░░░███████████████████░░░░░
  ███████████████████░░░░░░░█████████████████░░░░░░███████████████████░░░░░
  ███████████████████░░░░░░░░███████████████░░░░░░░██████████░░░░░░░░░░░░░░
  ███████████████████░░░░░░░░███████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
  ███████████████████░░░░░░░░███████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
  ███████████████████░░░░░░░░░██░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
  ███████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
  ██████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░  ░░░░░░░
      ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░      ░░░
            ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░    ░░░░░░░░░░░░░░░░░░
               ░░░░░░░░░░░░░░░░░░░░░░░░░░░░
                    ░░░░░░░░░░░░░░░░░
                       ░░░░░
```

> On Sep 11, 2024, Mistral released a new [Pixtral 12B](https://x.com/MistralAI/status/1833758285167722836), their first multimodal model. This guide shows how to use run and deploy this multimodal model on any cloud and Kubernetes.

Reviewer: ```suggestion
On Sep 11, 2024, Mistral released [Pixtral 12B](https://x.com/MistralAI/status/1833758285167722836), their first multimodal model. This guide shows how to use easily run and deploy this multimodal model on your own cloud or Kubernetes cluster.
```
"their first multimodal model supporting vision and text inputs"?

Reviewer 2: ```suggestion
On Sep 11, 2024, Mistral released [Pixtral 12B](https://x.com/MistralAI/status/1833758285167722836), their first multimodal model. This guide shows how to easily run and deploy this multimodal model on your own cloud or Kubernetes cluster.
```

## Run Pixtral on Any Cloud or Kubernetes

> 1. Install SkyPilot on your local machine:

Reviewer: link to install page

add `sky check`?

```bash

> pip install 'skypilot-nightly[all]'

Reviewer: Is stable ok? Or must use nightly?

```

2. Launch the model on any cloud or Kubernetes:
```bash
sky launch -c pixtral pixtral.yaml
```

![Pixtral on SkyPilot](https://i.imgur.com/afzNiaK.png)

3. Get the endpoint and send requests:
```bash
ENDPOINT=$(sky status --endpoint 8081 pixtral)

> curl --location http://$ENDPOINT/v1/chat/completions \

Reviewer: we don't need --location now?

        "tool_calls": []
      },
      "logprobs": null,
      "finish_reason": "stop",
      "stop_reason": null
    }
  ],
  "usage": {
    "prompt_tokens": 4457,
    "total_tokens": 4764,
    "completion_tokens": 307
  },
  "prompt_logprobs": null
}
```

</details>

## Scale Up Pixtral Endpoint as a Service

> 1. Start a service with SkyServe:

Reviewer: Link to docs

```bash
sky serve up -n pixtral pixtral.yaml
```

2. Get the endpoint and send requests:
```bash
ENDPOINT=$(sky serve status --endpoint pixtral)

> curl --location http://$ENDPOINT/v1/chat/completions \

Reviewer: need -L?

    -H 'Content-Type: application/json' \
    -H 'Authorization: Bearer token' \
    --data '{
        "model": "mistralai/Pixtral-12B-2409",
        "messages": [
        {
            "role": "user",
            "content": [
                {"type" : "text", "text": "Describe this image in detail please."},
                {"type": "image_url", "image_url": {"url": "https://s3.amazonaws.com/cms.ipressroom.com/338/files/201808/5b894ee1a138352221103195_A680%7Ejogging-edit/A680%7Ejogging-edit_hero.jpg"}},
                {"type" : "text", "text": "and this one as well."},
                {"type": "image_url", "image_url": {"url": "https://www.wolframcloud.com/obj/resourcesystem/images/a0e/a0ee3983-46c6-4c92-b85d-059044639928/6af8cfb971db031b.png"}}
            ]
        }],
        "max_tokens": 1024
    }' | jq .

> ```

Reviewer 2: Should we add response in a dropdown here?
